### PR TITLE
37 us as a channel admin i want to ban or mute user from the channel

### PR DESCRIPTION
**Pull request by cnstll**
# Description
WIP
- Massive refacto of chat components with a wide reorg and cleaning of query cache management
- Can ban users and atm users will be unbanned after 30s
- WIP on mute / unmute feature. Endpoints ok. Missing clean logic in front.

# Changes include

- [x] New feature (non-breaking change that adds functionality)
- [x] New Bug (breaking changes that add unexpected functionalities)
- [ ] Bugfix (non-breaking change that solves an issue)

# Breaking changes

- Chat linked components are broadly affected by this PR

# Checklist

- [x] I have assigned this PR to myself
- [x] I have added at least 1 reviewer
- [x] I have added the needed labels
- [x] I have linked this PR to an issue
- [x] I have linked this PR to a milestone
- [x] I have linked this PR to a project
- [ ] I have tested this code
- [ ] I have added / updated tests (unit / functionals / end-to-end / ...)
- [ ] I have updated the README and other relevant documents (guides...)
- [ ] I have added sufficient documentation both in code, as well as in the READMEs

# Additional comments

Please post additional comments in this section if you have them, otherwise delete it.

### CODE REVIEW

File: frontend/app/src/components/global-components/interface.tsx [hunk 1]
```diff
@@ -19,23 +19,28 @@ export enum channelRole {
   Admin = 'ADMIN',
   Owner = 'OWNER',
 }
+export enum channelActionType {
+  Ban = 'BAN',
+  Block = 'BLOCK',
```
Comment (llescure) [inline frontend/app/src/components/global-components/interface.tsx:24]: Block is not a channel action type

File: backend/app/src/channel/channel.service.ts [hunk 2]
```diff
@@ -481,6 +496,138 @@ export class ChannelService {
...
+            channelActionOnChannelId: channelId,
+          },
+        },
+        select: {
+          type: true,
+          channelActionTargetId: true,
+          channelActionTime: true,
+        },
+      });
+      console.log(moderationAction);
```
Comment (llescure) [inline backend/app/src/channel/channel.service.ts:518]: Delete console.log

File: backend/app/src/channel/channel.service.ts [hunk 3]
```diff
@@ -481,6 +496,138 @@ export class ChannelService {
...
+          AND: [
+            { channelActionTargetId: targetUserId },
+            {
+              channelActionOnChannelId: channelId,
+            },
+            { type: actionType },
+          ],
+        },
+      });
+      console.log(deletedChannelAction);
```
Comment (llescure) [inline backend/app/src/channel/channel.service.ts:541]: delete console log

File: backend/app/src/channel/channel.gateway.ts [hunk 4]
```diff
@@ -226,7 +220,46 @@ export class ChannelGateway {
...
+    @GetCurrentUserId() requesterId: string,
+    @MessageBody('banInfo') banInfo: ModerateChannelDto,
+    @ConnectedSocket() clientSocket: Socket,
+  ) {
+    const banResult = await this.channelService.banFromChannelWS(
+      requesterId,
+      banInfo,
+    );
+    console.log('Banned result: ', banResult);
+    if (banResult == null || typeof banResult === 'string') {
```
Comment (llescure) [inline backend/app/src/channel/channel.gateway.ts:233]: Why no === ?

File: backend/app/src/channel/channel.gateway.ts [hunk 5]
```diff
@@ -226,7 +220,46 @@ export class ChannelGateway {
...
+    @GetCurrentUserId() requesterId: string,
+    @MessageBody('muteInfo') muteInfo: ModerateChannelDto,
+    @ConnectedSocket() clientSocket: Socket,
+  ) {
+    const muteResult = await this.channelService.muteFromChannelWS(
+      requesterId,
+      muteInfo,
+    );
+    console.log('Muted result: ', muteResult);
+    if (muteResult == null || typeof muteResult === 'string') {
```
Comment (llescure) [inline backend/app/src/channel/channel.gateway.ts:252]: Why no === ?

File: backend/app/src/channel/channel.service.ts [hunk 6]
```diff
@@ -481,6 +505,126 @@ export class ChannelService {
...
+
+  async getListOfUsersUnderModerationInChannel(
+    channelId: string,
+    channelActionType: ChannelActionType,
+  ) {
+    const usersUnderModeration = await this.prisma.channelAction.findMany({
+      where: {
+        channelActionOnChannelId: channelId,
+        type: channelActionType,
+      },
```
Comment (llescure) [inline backend/app/src/channel/channel.service.ts:563]: Maybe add a select not to have all users info ?

File: frontend/app/src/components/section-components/chat/mute-user.tsx [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+  }
+
+  return (
+    <>
+      {isMutedQuery?.valueOf() ? (
+        <p
+          className="text-center hover:underline my-2 truncate cursor-pointer"
+          onClick={unMuteUser}
+        >
+          Unute
```
[the remark this answers is not in the document]
Comment (cnstll) [reply frontend/app/src/components/section-components/chat/mute-user.tsx]: Removed unmute functionality

File: backend/app/src/channel/channel.service.ts (hunk 2 above)
Comment (cnstll) [reply backend/app/src/channel/channel.service.ts]: done

File: backend/app/src/channel/channel.gateway.ts (hunk 4 above)
Comment (cnstll) [reply backend/app/src/channel/channel.gateway.ts:233]: added ===

**Review by llescure — APPROVED** [review]

**Pull request merged**